src/lib.rs

```diff
@@ -131,13 +131,9 @@ impl<T: AsMut<[u8]>> LogBuffer<T> {
             byte & 0b11111000 == 0b11110000
         }
 
-        let buffer = self.buffer.as_mut();
-        for i in 0..buffer.len() {
-            if is_utf8_leader(buffer[i]) {
-                return core::str::from_utf8(&buffer[i..]).unwrap()
-            }
-        }
-        return ""
+        let buffer = &*self.buffer.as_mut();
+        buffer.iter().cloned().position(is_utf8_leader).map_or("", |i|
```

Comment:
I see. To be honest I still feel like this change makes the code harder to understand. The code before it was very straightforward.

Your other changes are fine though, I'll cherry-pick them.